core/modules/qdisp/QdispPool.cc

```diff
@@ -0,0 +1,196 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2018 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "QdispPool.h"
+
+#include "lsst/log/Log.h"
+
+// Qserv headers
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.qdisp.QdispPool");
+}
+
+
+namespace lsst {
+namespace qserv {
+namespace qdisp {
+
+
+// must hold pq._mtx before calling
+std::ostream& operator<<(std::ostream& os, PriorityQueue const& pq) {
+    auto iter = pq._queues.begin();
+    auto end = pq._queues.end();
+    for (;iter != end; ++iter) {
+        PriorityQueue::PriQ::Ptr const& que = iter->second;
+        os << "(pri=" << que->getPriority()
+           << ":sz="  << que->size()
+           << ":r="   << que->running << ")";
+    }
+    return os;
+}
+
+
+///< @Return true if the queue could be added.
+bool PriorityQueue::addPriQueue(int priority, int minRunning) {
+    std::lock_guard<std::mutex> lock(_mtx);
+    auto q = std::make_shared<PriQ>(priority, minRunning);
+    std::pair<int, PriQ::Ptr> item(priority, q);
+    auto ret = _queues.insert(item);
+    if (!ret.second) {
```

Comment:
I'm not sure why that is better. insert() returns pair<iterator, bool> where the bool is true if a new element was inserted. There's no pointer and the map is only searched once.

Reply:
It's better because it's more obvious. In my proposal the intent is shown in the very trivial algorithm: 1) check if a queue for the specified priority is registered, and 2) and if not then register the one.

BTW, I'm not insisting the code has to be changed that way. My only concern is to make the code a bit easier to read and comprehend by others (for benefits of the long term support).